Lib/test/test_itertools.py

```diff
@@ -902,35 +902,11 @@ def __next__(self):
             (([2], [3]), [4]),
             ([4], [5]),
         ])
-        check({2}, [
```

Comment:
These tests are passed for equivalent Python implementation. So it is reasonable to expect them passing for any correct C implementation.

Originally the purpose of these tests was to test bugs with using borrowed references. After removing them we cannot be sure that the bugs will not return. If you need to remove them, then perhaps the bugs returned.